x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -40,80 +44,103 @@ export interface IndexPatternPersistedState {
   currentIndexPattern: string;
 
   columnOrder: string[];
-  columns: {
-    [columnId: string]: IndexPatternColumn;
-  };
+  columns: Record<string, IndexPatternColumn>;
 }
 
 export type IndexPatternPrivateState = IndexPatternPersistedState & {
-  indexPatterns: { [id: string]: IndexPattern };
+  indexPatterns: Record<string, IndexPattern>;
 };
 
-// Not stateful. State is persisted to the frame
-export const indexPatternDatasource: Datasource<
-  IndexPatternPrivateState,
-  IndexPatternPersistedState
-> = {
-  async initialize(state?: IndexPatternPersistedState) {
-    // TODO: Make fetch request to load indexPatterns from saved objects
-    if (state) {
+export function getIndexPatternDatasource(chrome: Chrome, toastNotifications: ToastNotifications) {
+  // Not stateful. State is persisted to the frame
+  const indexPatternDatasource: Datasource<IndexPatternPrivateState, IndexPatternPersistedState> = {
+    async initialize(state?: IndexPatternPersistedState) {
+      const indexPatternObjects = await getIndexPatterns(chrome, toastNotifications);
+      const indexPatterns: Record<string, IndexPattern> = {};
+
+      if (indexPatternObjects) {
+        indexPatternObjects.forEach((obj: Partial<IndexPattern>) => {
```

Comment:
Can't we just do
```
        indexPatternObjects.forEach((obj) => {
          indexPatterns[obj.id] = obj;
        });
```

Basically getting rid of types and `!` operator because it will be happy with inferred types. It doesn't give me any errors

Reply:
Yes, this works now.